### PR TITLE
Named Vomit

### PR DESCRIPTION
**Pull request by Bucovineanu**
## About The Pull Request

Small chance for vomit to have a name

## Why It's Good For The Game

Because it's funny

## Changelog
:cl:
add: Some vomit may be named after famous trash.
/:cl:

### CODE REVIEW

**Comment by AcapitalA** [comment]
Based grudge PR

**Comment by TheChosenEvilOne** [comment]
I would recommend leaving the default names out and instead just using the name json from the server for this.

**Comment by EdgeLordExe** [comment]
it'd be funnier if you could 'dedicate' the vomit to someone, changing it's name (picture a mime making a post-modernist art piece which is just a pile of vomit named after the captain)

**Comment by Bucovineanu** [comment]
I left the default names out and now you can dedicate them with a pen.

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 1]
```diff
@@ -137,6 +137,27 @@
 	random_icon_states = list("vomit_1", "vomit_2", "vomit_3", "vomit_4")
 	beauty = -150
 
+/obj/effect/decal/cleanable/vomit/Initialize()
+	. = ..()
+	if(prob(5))
+		var/json_file = file("[global.config.directory]/trashnames.json")
+		var/list/names = list()
+		if(fexists(json_file))
+			names = json_decode(file2text(json_file))
```
Comment (TheChosenEvilOne) [inline code/game/objects/effects/decals/cleanable/misc.dm:146]:
```suggestion
		if(!fexists(json_file))
			return
		names = json_decode(file2text(json_file))
```

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 2]
```diff
@@ -137,6 +137,27 @@
...
+			name = pick(names)
+
+/obj/effect/decal/cleanable/vomit/attackby(obj/item/I, mob/user, params)
+	..()
+	if(istype(I, /obj/item/pen))
+		var/t = sanitize_name(stripped_input(usr, "Enter new vomit name", name, "vomit",MAX_NAME_LEN), allow_numbers = TRUE)
+		if(!t)
+			return
+		if(!in_range(src, usr) && loc != usr)
+			return
```
Comment (TheChosenEvilOne) [inline code/game/objects/effects/decals/cleanable/misc.dm:157]:
```suggestion
		if(!t || !in_range(src, usr))
			return
```
you probably aren't going to keep vomit in your inventory.

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 3]
```diff
@@ -137,6 +137,27 @@
...
 
+/obj/effect/decal/cleanable/vomit/Initialize()
+	. = ..()
+	if(prob(5))
+		var/json_file = file("[global.config.directory]/trashnames.json")
+		var/list/names = list()
+		if(fexists(json_file))
+			names = json_decode(file2text(json_file))
+		if(names.len > 0)
+			name = pick(names)
```
Comment (TheChosenEvilOne) [inline code/game/objects/effects/decals/cleanable/misc.dm:148]:
```suggestion
		if(names.len)
			name = pick(names)
```

**Review by TheChosenEvilOne — CHANGES_REQUESTED** [review]

File: code/game/objects/effects/decals/cleanable/misc.dm [hunk 4]
```diff
@@ -137,6 +137,26 @@
...
+/obj/effect/decal/cleanable/vomit/Initialize()
+	. = ..()
+	if(prob(5))
+		var/json_file = file("[global.config.directory]/trashnames.json")
+		var/list/names = list()
+		if(!fexists(json_file))
+			return
+		names = json_decode(file2text(json_file))
+		if(names.len)
+			name = pick(names)
```
Comment (TheChosenEvilOne) [inline code/game/objects/effects/decals/cleanable/misc.dm:149]: now that I think about this, the vomit list should be loaded only once.

**Comment by TheChosenEvilOne** [comment]
mfw waiting for checks

**Pull request merged**